torch/csrc/jit/tensorexpr/eval.h

```diff
@@ -29,6 +29,18 @@ class Value {
     Intvalues.push_back(0);
   }
 
+  template <typename T>
+  Value(Dtype dtype, T v) : dtype_(dtype) {
+#define TYPE_CASE(Type, Name)  \
+  if (dtype == k##Name) {      \
```

Comment:
I don't remember details of the dtypes, but is there any difference between `dtype == k##Name` and `dtype.scalar_type() == ScalarType::Name`, which is used in lots of others places?

Reply:
Yeah, thk `k##Name`s are all lanes=1, whereas checking the scalar type could let through vectorized types.  The latter *does* look cleaner (I didn't actually think of it at the time) but it does actually feel more accurate to use `==k##Name` since these are scalar constructors.